base/set.jl

```diff
@@ -4,6 +4,7 @@ mutable struct Set{T} <: AbstractSet{T}
     dict::Dict{T,Void}
 
     Set{T}() where {T} = new(Dict{T,Void}())
+    Set{T}(s::Set{T}) where {T} = new(Dict{T,Void}(s.dict))
```

Comment:
Possibly simpler: `new(copy(s.dict))` ?

Reply:
Ah, I see that this would not be the same code path, and could maybe be less efficient.

Reply:
Also in the future `copy` might be redefined and this approach is self contained within the definition of `Dict` constructions.